scripts/new_python_based_image.py

```diff
@@ -0,0 +1,555 @@
+import argparse
+import os
+import shutil
+import sys
+import platform
+import subprocess
+import logging
+import re
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def configure_logger(log_level):
+    """
+    Configures the logging settings based on the provided log level.
+
+    Args:
+        log_level (str): The logging level to set (e.g., 'INFO', 'DEBUG').
+    """
+    level = getattr(logging, log_level.upper(), logging.INFO)
+    logging.basicConfig(
+        level=level,
+        format="[%(levelname)s] %(asctime)s: %(message)s",
+        datefmt="%H:%M:%S"
+    )
+    LOGGER.setLevel(log_level)
+
+
+class Args:
+    """
+    Class to encapsulate command-line arguments.
+
+    Attributes:
+        context_dir (str): The directory to use as the context for searching.
+        source (str): The source Python version.
+        target (str): The target Python version.
+        match (str): The string to match with the paths.
+        log_level (str): The logging level.
+    """
+
+    def __init__(self, context_dir, source, target, match, log_level):
+        self.context_dir = context_dir
+        self.source = source
+        self.target = target
+        self.match = match
+        self.log_level = log_level
+
+    def __str__(self):
+        return (f"Arguments:\n"
+                f"Context Directory:  {self.context_dir}\n"
+                f"Source Version:     {self.source}\n"
+                f"Target Version:     {self.target}\n"
+                f"Match:              {self.match}\n"
+                f"Log Level:          {self.log_level}")
+
+
+def extract_input_args():
+    """
+    Extracts and validates command-line arguments.
+
+    Returns:
+        Args: An instance of the Args class containing parsed arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description="Script to create a new Python-based image from an existing one.",
+        usage="python script.py --context-dir <directory> --source <python_version_source> --target <python_version_target> --match <match> [--log-level <level>]"
+    )
+
+    parser.add_argument(
+        "--context-dir",
+        help="The directory to be the context for searching.")
+    parser.add_argument(
+        "--source",
+        help="The Python version to base the new image from.")
+    parser.add_argument(
+        "--target",
+        help="The Python version to be used in the new image.")
+    parser.add_argument(
+        "--match",
+        help="The string to match with the paths to base the new image from.")
+    parser.add_argument(
+        "--log-level", default="INFO",
+        help="Set the logging level. Default: INFO.")
```

Comment:
```suggestion
        help="Set the logging level. Default: %(default)s.")
```

read all about it https://docs.python.org/3/library/argparse.html#help